cmd/bios/bios.go

```diff
@@ -18,24 +19,38 @@ type biosActionFlags struct {
 	serverID string
 }
 
-func (f *biosActionFlags) ToCondition() (*types.ConditionCreate, error) {
-	id, err := f.ParseServerID()
+func CreateBiosControlCondition(ctx context.Context, action rctypes.BiosControlAction) error {
+	theApp := mctl.MustCreateApp(ctx)
+
+	client, err := app.NewConditionsClient(ctx, theApp.Config.Conditions, theApp.Reauth)
+	if err != nil {
+		return err
+	}
+
+	serverID, err := biosFlags.ParseServerID()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	biosParams := rctypes.NewBiosControlTaskParameters(id, rctypes.ResetSettings)
+	params := rctypes.NewBiosControlTaskParameters(serverID, action)
 
-	params, err := json.Marshal(biosParams)
+	response, err := client.ServerBiosControl(ctx, params)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return &types.ConditionCreate{Parameters: params}, nil
+	conditionResp, err := mctl.ConditionFromResponse(response)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("status=%d msg=%s conditionID=%s", response.StatusCode, response.Message, conditionResp.ID)
+
+	return err
 }
 
 func (f *biosActionFlags) ParseServerID() (uuid.UUID, error) {
```

Comment:
Why are we doing this in an external function? Does this syntactic sugar buy us anything?

Reply:
Ill remove it in my next PR.